core/trino-main/src/main/java/io/trino/server/protocol/PreparedStatements.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.protocol;
+
+import io.airlift.compress.zstd.ZstdCompressor;
+import io.airlift.compress.zstd.ZstdDecompressor;
+
+import static com.google.common.io.BaseEncoding.base64Url;
+import static java.lang.Math.toIntExact;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public final class PreparedStatements
+{
+    private PreparedStatements() {}
+
+    // No valid SQL statement starts with $
+    private static final String PREFIX = "$zstd:";
+
+    // TODO bigger value; low for testing
```

Comment:
Make these configurable in `ProtocolConfig`. This will also allow disabling if necessary, in case someone has a compatibility issue. You can inject this class into both of the calling classes.